app/src/main/kotlin/com/wire/android/ui/calling/ongoing/participantsview/ParticipantTile.kt

```diff
@@ -277,32 +282,73 @@ private fun AvatarTile(
 private fun UsernameTile(
     modifier: Modifier,
     name: String,
-    isSpeaking: Boolean
+    isSpeaking: Boolean,
+    hasEstablishedAudio: Boolean
 ) {
     val color = if (isSpeaking) MaterialTheme.wireColorScheme.primary else Color.Black
 
-    Surface(
-        modifier = modifier,
-        shape = RoundedCornerShape(dimensions().corner4x),
-        color = color
-    ) {
-        Text(
-            color = Color.White,
-            style = MaterialTheme.wireTypography.label01,
-            modifier = Modifier.padding(dimensions().spacing4x),
-            text = name,
-            maxLines = 1,
-            overflow = TextOverflow.Ellipsis
-        )
+    ConstraintLayout(modifier = modifier) {
+        val (nameLabel, connectingLabel) = createRefs()
+
+        Surface(
+            modifier = Modifier.constrainAs(nameLabel) {
+                bottom.linkTo(parent.bottom)
+                start.linkTo(parent.start)
+                end.linkTo(connectingLabel.start)
+            },
+            shape = RoundedCornerShape(
+                topStart = dimensions().corner4x,
+                bottomStart = dimensions().corner4x,
+                topEnd = if (hasEstablishedAudio) dimensions().corner4x else 0.dp,
+                bottomEnd = if (hasEstablishedAudio) dimensions().corner4x else 0.dp,
+            ),
+            color = color
+        ) {
+            Text(
+                color = Color.White,
+                style = MaterialTheme.wireTypography.label01,
+                modifier = Modifier.padding(dimensions().spacing4x),
+                text = name,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+        }
+        if (!hasEstablishedAudio) {
+            Surface(
```

Comment:
Does it need to be wrapped in `Surface`? The `Surface` above I believe is added mainly to create this background with rounded corners, so here it's not needed as both `Text`s are inside this parent `Surface`

Reply:
No, they are two different block each one has it's own text and corner